fuzzing/tools/launcher.py

```diff
@@ -18,58 +18,85 @@
 behind a number of arbitrary fuzzing engines.
 """
 
+import os
+import shutil
+import sys
+
 from absl import app
 from absl import flags
-import os
 
 FLAGS = flags.FLAGS
 
+flags.DEFINE_string(
+    "engine_launcher", None,
+    "Path to a shell script that launches the fuzzing engine executable with the appropriate command line arguments."
+)
+
+flags.DEFINE_string("binary_path", None, "Path to the fuzz test binary.")
+
 flags.DEFINE_bool(
     "regression", False,
     "If set True, the script will trigger the target as a regression test.")
 
-flags.DEFINE_enum(
-    "engine", "default", ["default", "libfuzzer"],
-    "The type of the engine, the default is to run a gUnit test.")
-
 flags.DEFINE_integer(
     "timeout_secs",
     0,
     "The maximum duration, in seconds, of the fuzzer run launched.",
     lower_bound=0)
 
-flags.DEFINE_list(
-    "fuzzer_extra_args", None,
-    "If non-empty, the elements will be passed to the fuzzer as arguments.")
-
 flags.DEFINE_string(
-    "corpus_dir", "",
+    "corpus_dir", None,
     "If non-empty, a directory that will be used as a seed corpus for the fuzzer."
 )
 
-flags.DEFINE_string("dict", "",
+flags.DEFINE_string("dictionary_path", None,
                     "If non-empty, a dictionary file of input keywords.")
 
+flags.DEFINE_string(
+    "fuzzing_output_root", "/tmp/fuzzing",
+    "The root directory for storing all generated artifacts during fuzzing.")
+
+flags.DEFINE_bool(
+    "clean", False,
+    "If set, cleans up the output directory of the target before fuzzing.")
+
+flags.mark_flag_as_required("engine_launcher")
+flags.mark_flag_as_required("binary_path")
+
 
 def main(argv):
-    if len(argv) != 2:
-        raise app.UsageError(
-            "This script receives 1 argument. It should look like:" +
-            "\n\tpython " + __file__ + " EXECUTABLE")
-
-    command_args = [argv[1]]
-    if FLAGS.engine == "libfuzzer":
-        command_args.append("-max_total_time=" + str(FLAGS.timeout_secs))
-        command_args.append("-timeout=" + str(FLAGS.timeout_secs))
-        if FLAGS.regression:
-            command_args.append("-runs=0")
-        if FLAGS.dict:
-            command_args.append("-dict=" + FLAGS.dict)
+    # TODO(sbucur): Obtain a target-specific path here.
```

Comment:
The idea is that all fuzzing engines are using the environment vars set by this common "pre-"launcher, right? 

Struggling to see where this is tied to cc_fuzzing_engine right now, is this a TODO? (please help!)

Reply:
Yeah, this is a bit confusing!

Everything should be already working actually. The `cc_fuzzing_engine` rule receives a launcher shell script, which is passed to this launcher script by the `cc_fuzz_test` rule, through this script's `--engine_launcher` flag.

You can think of this Python script as the front-end, which is concerned with a uniform flag interface for the developer. The launcher (shell) script is the back-end, which adapts the fuzzer run configuration to the engine-specific flag interface.

I wrote a more detailed docstring at the top of this file, and also started a documentation stub in the README file with the environment variables available to the launcher script. PTAL.

Reply:
AH! Thank you, finally tracked this down for myself. It didn't show `default = Label("//fuzzing/tools:launcher"),` because it wasn't in the diff. Thank you!